### ISSUE TITLE
Upgrade to use isPromise from rxjs

### ISSUE THREAD

**Issue by ChrisDahmCS** [angular/angular#27667]
# 🚀 feature request

### Relevant Package
This feature request is for @angular/angular [common, compiler, core, forms, router]

### Description
lang.ts has the isPromise method, but rxjs already provides this. Upgrade to use the provided method from rxjs.

### Describe the solution you'd like
Refactor the custom isPromise method to the [isPromise method provided by rxjs](https://github.com/ReactiveX/rxjs/blob/master/src/internal/util/isPromise.ts).

This is similar to [this TODO](https://github.com/angular/angular/blob/master/packages/core/src/util/lang.ts#L24).

### Describe alternatives you've considered
None

### Additional Notes

I'd be happy to do the PR :)

**Comment by mhevery** [comment]
Why is rxjs method better than the one in angular. Such a change would make angular depend more on RXJS which I don't think is a good direction.

**Comment by ChrisDahmCS** [comment]
@mhevery so is this case different than isObservable because Promise is not sourced by rxjs?

**Comment by pkozlowski-opensource** [comment]
I agree with @mhevery that such change would introduce unnecessary, accidental coupling. IMO we should keep the integration surface as small as possible.

I've got also reservations about using `isObservable` from RxJS but it is another topic.

Closing as won't fix.
